plugins/autowarmth.koplugin/main.lua

```diff
@@ -0,0 +1,780 @@
+            help_text = _("Easy mode disables all entries except: civil dawn, sun rise, sun set and civil dusk."),
+            callback = function(touchmenu_instance)
+                local info_text
+                if not self.easy_mode then
+                    info_text = _("Reduce entries in auto-warmth schedule?")
+                else
+                    info_text = _("Expand entries in auto-warmth schedule?")
+                end
+                UIManager:show(ConfirmBox:new{
+                    text = info_text,
+                    ok_callback = function()
+                        self.easy_mode = not self.easy_mode
+                        G_reader_settings:saveSetting("autowarmth_easy_mode", self.easy_mode)
+                        self:scheduleMidnightUpdate()
+                        -- closing menu is necessary for refreshing the menu structure
+                        if touchmenu_instance then touchmenu_instance:closeMenu() end
+                    end,
+                })
+            end,
+            keep_menu_open = true,
+        },
+        {
+            text = _("Activate"),
+            checked_func = function()
+                return self.activate ~= 0
+            end,
+            sub_item_table = self:getActivateMenu(),
+        },
+        {
+            text = _("Location"),
+            enabled_func = function() return self.activate ~= activate_schedule end,
+            sub_item_table = self:getLocationMenu(),
+        },
+        {
+            text = _("Schedule"),
+            enabled_func = function() return self.activate ~= activate_sun end,
+            sub_item_table = self:getScheduleMenu(),
+        },
+        {
+            text = _("Warmths"),
+            sub_item_table = self:getWarmthMenu(),
+            separator = true,
+        },
+        self:getTimesMenu(_("Active auto-warmth parameters")),
+        self:getTimesMenu(_("Infos about the sun in"), true, activate_sun),
+        self:getTimesMenu(_("Infos about the schedule"), false, activate_schedule),
+    }
+end
+
+function AutoWarmth:getActivateMenu()
+    local function getActivateMenuEntry(text, activator)
+        return {
+            text = text,
+            checked_func = function() return self.activate == activator end,
+            callback = function()
+                if self.activate ~= activator then
+                    self.activate = activator
+                else
+                    self.activate = 0
+                end
+                G_reader_settings:saveSetting("autowarmth_activate", self.activate)
+                self:scheduleMidnightUpdate()
+            end,
+        }
+    end
+
+    return {
+        getActivateMenuEntry( _("Sun position"), activate_sun),
+        getActivateMenuEntry( _("Time schedule"), activate_schedule),
+        getActivateMenuEntry( _("Whatever is closer to noon"), activate_closer_noon),
+        getActivateMenuEntry( _("Whatever is closer to midnight"), activate_closer_midnight),
```

Comment:
> One wording I want to change in the activation menu:
'Whatever is closer to 12:00' and 'Whatever is closer to 24:00'
This is important as 'high noon' is really meant to be the highest stand of the sun and midnight is +- 12 hours from that.

I'm not sure I quite follow. Is there a reason you didn't just write something like "solar noon" or "highest point of the sun in the sky"?

Reply:
Yes I will use "noon" for 12:00 and "midnight" for 24:00, and change "high noon" to "solar noon" and midnight to "solar midnight". Then it should be clear.